File: Programmers/연습문제/Level 1/K번째수/Solution.js
function solution(array, commands) {
    const answer = [];
    for (const command of commands) {
        const [start, end, k] = command;
        const slicedArray = sliceArray(array, start, end);
        const kthNum = getSortedKth(slicedArray, k);
        answer.push(kthNum);
    }
    return answer;
}

function sliceArray(arr, start, end) {
    return arr.slice(start-1, end);
}

function getSortedKth(arr, k) {
    arr.sort((a, b) => a - b);
    return arr[k-1];
}


// TC
const array = [1, 5, 2, 6, 3, 7, 4];
const commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]];
console.log(solution(array, commands)); // [5, 6, 3]

File: Programmers/연습문제/Level 1/K번째수/Solution.swift
import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    return commands.map {
        let (i, j, k) = ($0[0]-1, $0[1]-1, $0[2]-1)
        return array[i...j].sorted()[k]
    }
}

// TC
let array = [1, 5, 2, 6, 3, 7, 4]
let commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
print(solution(array, commands)) // [5, 6, 3]
